AutoCollection/AutoCollector.ts

```diff
@@ -0,0 +1,171 @@
+import AutoCollectConsole = require("./Console");
+import AutoCollectExceptions = require("./Exceptions");
+import AutoCollectPerformance = require("./Performance");
+import AutoCollecPreAggregatedMetrics = require("./PreAggregatedMetrics");
+import HeartBeat = require("./HeartBeat");
+import AutoCollectHttpDependencies = require("./HttpDependencies");
+import AutoCollectHttpRequests = require("./HttpRequests");
+import TelemetryClient = require("../Library/NodeClient");
+import { AutoCollectNativePerformance, IDisabledExtendedMetrics } from "./NativePerformance";
+import { JsonConfig } from "../Library/JsonConfig";
+
+class AutoCollector {
+
+    // Default values
+    public isConsole = true;
+    public isConsoleLog = false;
+    public isExceptions = true;
+    public isPerformance = true;
+    public isPreAggregatedMetrics = true;
+    public isHeartBeat = false;
+    public isRequests = true;
+    public isDependencies = true;
+    public isNativePerformance = true;
+    public isCorrelating = true;
+    public forceClsHooked: boolean;
+    public disabledExtendedMetrics: IDisabledExtendedMetrics;
+
+    private _console: AutoCollectConsole;
+    private _exceptions: AutoCollectExceptions;
+    private _performance: AutoCollectPerformance;
+    private _preAggregatedMetrics: AutoCollecPreAggregatedMetrics;
+    private _heartbeat: HeartBeat;
+    private _nativePerformance: AutoCollectNativePerformance;
+    private _serverRequests: AutoCollectHttpRequests;
+    private _clientRequests: AutoCollectHttpDependencies;
+    private _client: TelemetryClient;
+    private _isStarted = false;
+
+    public setup(client: TelemetryClient) {
+        this._client = client;
```

Comment:
Compared the old code, what if the client already setup? Do we not need the logging info anymore?

Reply:
Never mind... I get it, the appinsights.ts setup keeps the previous logic.